pallets/proofs-dealer/src/lib.rs

```diff
@@ -840,6 +887,30 @@ pub mod pallet {
                 T::BlockFullnessPeriod::get(),
                 T::ChallengeTicksTolerance::get()
             );
+
+            // Check that if `ChallengesFee` is greater than 0, then `ChallengeOrigin` cannot be root.
+            // This prevents the misconfiguration where a fee is charged but the origin is root (which cannot be charged).
+            if !T::ChallengesFee::get().is_zero() {
+                // Test that root origin is rejected by ChallengeOrigin
+                let root_origin = frame_system::RawOrigin::Root.into();
+                assert!(
+                    T::ChallengeOrigin::try_origin(root_origin).is_err(),
+                    "ChallengeOrigin cannot be root when ChallengesFee ({:?}) is greater than 0, as root cannot be charged fees",
+                    T::ChallengesFee::get()
+                );
+            }
+
+            // Check that if `PriorityChallengesFee` is greater than 0, then `PriorityChallengeOrigin` cannot be root.
+            // This prevents the misconfiguration where a fee is charged but the origin is root (which cannot be charged).
+            if !T::PriorityChallengesFee::get().is_zero() {
+                // Test that root origin is rejected by PriorityChallengeOrigin
+                let root_origin = frame_system::RawOrigin::Root.into();
+                assert!(
+                    T::PriorityChallengeOrigin::try_origin(root_origin).is_err(),
+                    "PriorityChallengeOrigin cannot be root when PriorityChallengesFee ({:?}) is greater than 0, as root cannot be charged fees",
+                    T::PriorityChallengesFee::get()
+                );
+            }
```

Comment:
If it's possible, we can replace these checks with if the fees > 0, then check the `origin.into_signer` returns a `Some(_account_id)`.